src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordAsTopLevel.java

```diff
@@ -0,0 +1,57 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+/* Config:
+ * checkFields = false
+ * checkMethods = true
+ * validateOnlyOverlapping = false
+ *
+ */
+public record InputRequireThisRecordAsTopLevel(int x, int y) {
```

Comment:
Please add some usages of the record components (`x` or `y`) to the compact ctor and any method.
All record componens should be passed to `addInstanceMember` method, but do not see such changes in this commit.

Reply:
Good catch, I have fixed this and added some more test inputs.